### ISSUE TITLE
Test ToMany function fails due to unique column randomization not being adhered to

### ISSUE THREAD

**Issue by tiger5226** [volatiletech/sqlboiler#277]
If you're having a generation problem please answer these questions before submitting your issue. Thanks!

### What version of SQLBoiler are you using (`sqlboiler --version`)?

latest

### If this happened at generation time what was the full SQLBoiler command you used to generate your models? (if not applicable leave blank)

sqlboiler --no-auto-timestamps --no-hooks --tinyint-as-bool --wipe mysql

### If this happened at runtime what code produced the issue? (if not applicable leave blank)

N/A

### What is the output of the command above with the `-d` flag added to it? (Provided you are comfortable sharing this, it contains a blueprint of your schema)

No errors

### Please provide a relevant database schema so we can replicate your issue (Provided you are comfortable sharing this)

Can't share schema

### Further information. What did you do, what did you expect?

So if you look at the test code generation template file below

https://github.com/volatiletech/sqlboiler/blob/master/templates_test/relationship_to_many_setops.tpl#L19

The columns that it takes as input are the primary key columns, and columns without defaults. The problem here is that only the primary key is considered as being unique. It does not consider that columns other than the primary key might need to be unique. 

This obviously causes test case failures which impedes the auto generation capabilities if they have to be adjusted after each generation.

**Comment by ceshihao** [comment]
Do you mean you have a non-primary-key and unique column with default value?

It seems not to be very reasonable.

**Comment by tiger5226** [comment]
So I have 2 columns, the id (serial) and another address. The uniqueness of a row cannot be a composite primary key. Therefore the standard is to use a unique constraint in addition to the primary key in mysql...as far as I understand. Also this is common to improve query performance when you know a column is unique.

**Comment by aarondl** [comment]
I mean, it doesn't matter what you're using the unique column for. In general we should be able to have unique columns and not have them die during testing. However what's confusing to me is that the whole purpose of the random seed is that it is an ever-increasing integer that acts as the seed for each column's value. Which means that unless this seed isn't being incremented, or the value range of your unique column is extremely constrained (like it's a boolean), it shouldn't be failing? Given this information, can you at least tell me the type of column that's failing so I can look into the problem in the `randomize` package?

**Comment by aarondl** [comment]
It'd be helpful to know both the type and any constraints other than unique on it.

**Comment by tiger5226** [comment]
Yes absolutely. [Schema](https://github.com/lbryio/chainquery/blob/master/db/chainquery_schema.sql#L5). The link goes to the specific line. Although, i have seen this problem in other areas as well. It is actually a bit random too. Sometimes the tests fail sometimes they don't. I have removed the tests for now to make sure the travis build works. 

Lastly, I have to do some cleanup on the schema, there are some duplicate indices.

**Comment by aarondl** [comment]
Hey Tiger. I was about to get working on this but I'm currently side tracked on another issue and it's taking forever. I wanted to make a suggestion.

You said that sometimes the tests fail and sometimes they pass. So the only thing I can think of because we use the random `Seed` which is ever-incrementing, it depends on maybe which columns are getting randomized randomly and which are not. Or which are null or what have you is what's the probably problem here.

But I also had a sneaking suspicion maybe the transaction isolation level isn't good enough and it's running into collisions with other tests inserting into that table? Maybe you can deparallelize the test suite entirely and see if you still have trouble. I think the flag is `go test -parallel 1`. If you also use `-v` you'll be able to make sure nothing's being run in parallel.

**Comment by tiger5226** [comment]
I can try that out. I had a strong suspicion of this as well.

**Comment by tiger5226** [comment]
hmm, no luck. When I created the tests again, the build failed during testing and another issue related to the NewConfig change. I wonder if that has to do with the file generation. I have some things to get done tonight but I will try to debug and clear that up soon to test it out.

**Comment by aarondl** [comment]
Got to this a little bit today actually but I'm not able to reproduce the error being seen here. I used a few tables from the schema provided.
